torch/fx/experimental/proxy_tensor.py

```diff
@@ -1371,6 +1372,38 @@ def __torch_function__(
             return node
             # Don't actually run the function! We just want to trace the calls
             # into a graph. We don't actualy want to change global autograd state.
+        if func in [torch._C._functorch._add_batch_dim, torch._C._functorch._remove_batch_dim]:
+            # Don't actually run the function! We just want to trace the calls
+            # into a graph. We don't actualy want to change global functorch state.
+
+            def fetch_proxy(x):
```

Comment:
hmm this feels like a lot of boilerplate:

(1) are these functions below identical to the ones we're calling from inside the `ProxyTorchDispatchMode`? it might be cleaner to split this into a shared util function or two that we can call from both places

(2) can we just unconditionally run the logic to grab input proxies / set output proxies using `pytree.tree_map_only(torch.Tensor)`? that should automatically handle ops that do/do not have tensor inputs or outputs